node_modules/col-whiteboards/lib/api.js

```diff
@@ -0,0 +1,771 @@
+    notifyOnlineUsers(socket.whiteboard.id);
+
+    return callback();
+  });
+};
+
+/**
+ * Delete a whiteboard session for a specific websocket connection
+ *
+ * @param  {String}         socketId                        The unique socket id of the whiteboard session that is being deleted
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var deleteWhiteboardSession = function(socketId, callback) {
+  callback = callback || function() {};
+
+  var validationSchema = Joi.object().keys({
+    'socketId': Joi.string().required()
+  });
+
+  var validationResult = Joi.validate({
+    'socketId': socketId
+  }, validationSchema);
+
+  if (validationResult.error) {
+    return callback({'code': 400, 'msg': validationResult.error.details[0].message});
+  }
+
+  // Get the whiteboard session that is being deleted
+  var options = {
+    'where': {
+      'socket_id': socketId
+    }
+  };
+
+  DB.WhiteboardSession.find(options).complete(function(err, whiteboardSession) {
+    if (err) {
+      log.error({'err': err, 'id': id, 'socketId': socketId}, 'Failed to retrieve the whiteboard session to delete');
+      return callback({'code': 500, 'msg': err.message});
+    } else if (!whiteboardSession) {
+      log.debug({'err': err, 'id': id, 'socketId': socketId}, 'The whiteboard session that is being deleted could not be found');
+      return callback({'code': 404, 'msg': 'The whiteboard session that is being deleted could not be found'});
+    }
+
+    // Delete the whiteboard session from the DB
+    whiteboardSession.destroy().complete(function(err) {
+      if (err) {
+        log.error({'err': err, 'socketId': socketId}, 'Failed to delete a whiteboard session');
+        return callback({'code': 500, 'msg': err.message});
+      }
+
+      // Send the updated list of online users to the users in the whiteboard
+      notifyOnlineUsers(whiteboardSession.whiteboard_id);
+
+      return callback();
+    });
+  });
+};
+
+/**
+ * Delete all whiteboard sessions. This is necessary when the server restarts, as all
+ * websockets will have disconnect without the disconnect logic being executed
+ *
+ * @param  {Function}       [callback]                      Standard callback function
+ * @param  {Object}         [callback.err]                  An error that occurred, if any
+ * @api private
+ */
+var deleteWhiteboardSessions = function(callback) {
+  callback = callback || function() {};
+
+  var options = {
+    'truncate': true
+  };
+
+  DB.WhiteboardSession.destroy(options).complete(function(err) {
+    if (err) {
+      log.error({'err': err}, 'Failed to delete all whiteboard session');
+      return callback({'code': 500, 'msg': err.message});
+    }
+
+    return callback();
+  });
+};
+
+// Delete all whiteboard sessions when the server starts
```

Comment:
I've outscoped multi-appserver support from this iteration, as the service will probably be run like that. However, if you think it's important to give it some consideration or if you think it'll be hard to retrofit further down the line, we can give it some thought now.